state-synchronizer/src/tests/integration_tests.rs

```diff
@@ -296,24 +297,39 @@ impl SynchronizerEnv {
             let authentication_mode =
                 AuthenticationMode::Mutual(self.network_keys[new_peer_idx].clone());
             let pub_key = authentication_mode.public_key();
-            let mut network_builder = NetworkBuilder::new(
-                self.runtime.handle().clone(),
-                ChainId::default(),
+            let network_context = Arc::new(NetworkContext::new(
                 self.network_id.clone(),
                 RoleType::Validator,
                 self.peer_ids[new_peer_idx],
+            ));
+            let mut network_builder = NetworkBuilder::new(
+                ChainId::default(),
+                trusted_peers.clone(),
+                network_context,
                 addr.clone(),
                 authentication_mode,
             );
             network_builder
-                .trusted_peers(trusted_peers)
-                .seed_peers(seed_peers)
-                .add_connectivity_manager()
-                .add_gossip_discovery(addr, constants::DISCOVERY_INTERVAL_MS, pub_key);
+                .add_connectivity_manager(
+                    seed_peers,
+                    trusted_peers,
+                    constants::MAX_FULLNODE_CONNECTIONS,
+                    constants::MAX_CONNECTION_DELAY_MS,
+                    constants::CONNECTIVITY_CHECK_INTERNAL_MS,
+                    constants::NETWORK_CHANNEL_SIZE,
+                )
+                .add_gossip_discovery(
+                    GossipConfig {
+                        advertised_address: addr,
+                        discovery_interval_ms: constants::DISCOVERY_INTERVAL_MS,
```

Comment:
this kinda interesting that they aren't uniform... why reveal the config outside the internals of the builder?

Reply:
+1

One of the big outstanding TODOS is to make add_gossip *private* and for this test (and for dummy.rs) to use a NodeConfig and ::create to instantiate the builder rather than reaching inside to do it manually.

(those manual reach-ins are actually pretty bad, and encourage some rather unfortunate practices and corner case support)